source/MaterialXCore/Element.h

```diff
@@ -117,6 +117,13 @@ class Element : public std::enable_shared_from_this<Element>
     ///    the returned path will be relative to this ancestor.
     string getNamePath(ConstElementPtr relativeTo = nullptr) const;
 
+    /// Return the element referred to by the path relative to the current
+    /// element. A path of empty string returns a pointer to the current
+    /// element.
+    /// @param path The path to use to find an element relative to the
+    ///    current element.
+    ElementPtr getElementByPath(const std::string& path);
```

Comment:
As already discussed this sounds more like getting a child or a descendent of this element which seems less ambiguous. Also should this method not be const since it doesn't modify it's own contents?

Reply:
It's returning an ElementPtr though, not a ConstElementPtr, so it can't be const.

Reply:
I agree "child" would be better, and then to also change the behavior above to return nullptr instead of itself if a matching child is not found. In order to follow the existing getNamePath method maybe it could be named getChildByNamePath in that case?